test/basic.test.js

```diff
@@ -625,3 +625,31 @@ test('Should support plugin level prefix', t => {
     })
   })
 })
+
+test('register the middleware at a different hook', async t => {
+  const fastify = Fastify()
+  t.teardown(fastify.close)
+
+  let onRequestCalled = false
+
+  await fastify.register(middiePlugin, {
+    hook: 'preHandler'
+  })
+
+  fastify.use(function (req, res, next) {
+    t.ok(onRequestCalled)
+    next()
+  })
+
+  fastify.addHook('onRequest', function (req, reply, next) {
+    onRequestCalled = true
+    next()
+  })
+
+  fastify.get('/', async (req, reply) => {
+    return { hello: 'world' }
+  })
```

Comment:
What part of this test is showing `middie` executing during the `preHandler` phase?

Reply:
The fact that it is executed _after_ the `onRequest` has fired. I know it's a bit indirect.